compiler/main/driver.cpp

```diff
@@ -506,6 +508,24 @@ static void setupChplHome(const char* argv0) {
   }
 }
 
+#ifndef HAVE_LLVM
+
+// If the compiler was built without LLVM and CHPL_LLVM is not set in
+// the environment, set it to 'none' in order to avoid having the
+// chplenv scripts potentially infer it to be 'system' or 'bundled',
+// which can only result in an error.
+//
+static void setupChplLLVM(void) {
+  // set CHPL_LLVM to 'none' if it isn't already set
+  if (getenv("CHPL_LLVM") == NULL) {
+    if (setenv("CHPL_LLVM", "none", 0) != 0) {
+      INT_FATAL("Problem setting CHPL_LLVM");
+    }
+    setChplLLVM = true;
```

Comment:
Is there any conflict here if the user throws `--llvm` (or maybe that's been replaced recently, so whatever has replaced it.) I think it's a user error if they're throwing `--llvm` for a compiler built without it, but I'm not sure if it changes the error message in an unhelpful way.

Reply:
Ah, based on the PR description and the code you updated in `checkLLVMCodeGen`, this code must be called after that `checkLLVMCodeGen`, so I think nevermind this question.

Reply:
`--llvm` is deprecated but still supported for this release, so throwing it results in a deprecation error and my new one:

```
warning: --llvm is deprecated -- please use --target-compiler=llvm
error: You have requested 'llvm' as the target compiler, but this copy of
       'chpl' was built without LLVM support.  Either select a different
       target compiler or re-build your compiler with LLVM enabled.
```

Similarly, using `--target-compiler=llvm` (or the equivalent CHPL_TARGET_COMPILER env var) results in:

```
error: You have requested 'llvm' as the target compiler, but this copy of
       'chpl' was built without LLVM support.  Either select a different
       target compiler or re-build your compiler with LLVM enabled.
```

Does that match what you were imagining / hoping for?

Reply:
Yup, thanks.